storage/spanstore/memory/memory.go

```diff
@@ -65,6 +65,9 @@ func (m *Store) GetDependencies(endTs time.Time, lookback time.Duration) ([]mode
 			for _, s := range trace.Spans {
 				parentSpan := m.findSpan(trace, s.ParentSpanID)
 				if parentSpan != nil {
+					if parentSpan.Process.ServiceName == s.Process.ServiceName {
```

Comment:
Other stores do not do the calculations themselves, they would simply return the links already computed externally.